dev/integration_tests/platform_interaction/lib/src/system_navigation.dart

```diff
@@ -0,0 +1,22 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'package:flutter/services.dart';
+import 'test_step.dart';
+
+Future<TestStepResult> systemNavigatorPop() async {
+  const BasicMessageChannel<String> channel = const BasicMessageChannel<String>(
+    'SystemNavigator.pop',
+    const StringCodec(),
+  );
+
+  TestStatus status = TestStatus.failed;
+  channel.setMessageHandler((String message) async {
+    status = TestStatus.ok;
```

Comment:
The test step result might be created before you get the message here. Maybe complete a `Completer<TestStepResult>` here instead, and return its `future` in line 21.